### PR TITLE
Add drafts link to profile page

### PR DESCRIPTION
**Pull request by TildaDares**
Part of larger planning issue #9667 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

Screenshot 2021-06-22 at 12 16 20

### CODE REVIEW

**Comment by jywarren** [comment]
Hi @TildaDares - do you want to check the logs to see what failed here? It's extra helpful if you can paste the error into a comment, if you don't immediately see how to fix it i'm happy to help!

File: app/models/user.rb [hunk 1]
```diff
@@ -517,6 +517,16 @@ def self.recently_active_users(limit = 5, order = 'last_updated DESC')
     end
   end
 
+  def drafts
+    Node.where(uid: uid)
```
Comment (jywarren) [inline app/models/user.rb:521]: nice!!

File: app/models/user.rb [hunk 2]
```diff
@@ -517,6 +517,16 @@ def self.recently_active_users(limit = 5, order = 'last_updated DESC')
...
+    .where(status: 3, type: 'note')
+    .order('created DESC')
+  end
+
+  def drafts_count
+    drafts.size
```
Comment (jywarren) [inline app/models/user.rb:527]: Couldn't the view code just run `user.drafts.size`?

File: app/views/users/profile.html.erb [hunk 3]
```diff
@@ -265,6 +265,9 @@
       <li><h5><a href="/wikis/author/<%= @profile_user.name %>"><%= pluralize(@profile_user.revisions.count, 'wiki edit', plural: 'wiki edits') %></a></h5></li>
       <li><h5><a href = "/tag/question:*/author/<%= params[:id] %>"><%= pluralize(Node.questions.where(status: 1, uid: @profile_user.id).length, 'question', plural:'questions') %></a></h5></li>
       <li><h5><a href = "/profile/comments/<%= params[:id] %>"><%= pluralize(Comment.where(uid: @profile_user.id).count, 'comment', plural: 'comments') %></a></h5></li>
+      <% if current_user&.can_moderate? || @profile_user == current_user %>
+        <li><h5><a href='#'><%= pluralize(@profile_user.drafts_count, 'draft', plural: 'drafts' )%></a></h5></li>
```
Comment (jywarren) [inline app/views/users/profile.html.erb:269]: awesome!

File: test/unit/user_test.rb [hunk 4]
```diff
@@ -309,4 +309,9 @@ class UserTest < ActiveSupport::TestCase
     user = users(:bob)
     assert_equal  user.subscriptions(:tag).size, 4
   end
+
+  test 'user drafts' do
+    user = users(:jeff)
+    assert_not user.drafts.empty?
```
Comment (jywarren) [inline test/unit/user_test.rb:315]: Love the unit test! Thank you!!!

**Review by jywarren — COMMENTED** [review]
This looks super so far. Are you planning to add the link to point somewhere in this PR or in a follow-up? I did note possibly `drafts_count` may not be entirely needed?

But great work, thanks Tilda!!!

File: app/models/user.rb (hunk 2 above)
Comment (TildaDares) [reply app/models/user.rb:527]: You are right!

**Comment by TildaDares** [comment]
> This looks super so far. Are you planning to add the link to point somewhere in this PR or in a follow-up?

The link will be in a follow-up PR. I'm still waiting on the PR for the author's notes template.

**Comment by RuthNjeri** [comment]
Great job @TildaDares

**Comment by jywarren** [comment]
Great, thank you!!!

**Pull request merged**